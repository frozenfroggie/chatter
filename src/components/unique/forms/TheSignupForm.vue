<template>
  <div class="inputContainer">
    <div :class="['username', {invalid: $v.username.$error && username !== ''}]">
      <icon name="user" scale="1"></icon>
      <input
        type="text"
        name="username"
        placeholder="Username"
        aria-label="Username"
        id="username"
        :value="username"
        @input="(event) => updateSignupForm(event, 'username')"
        @blur="$v.username.$touch()">
    </div>
    <div :class="['email', {invalid: $v.email.$error && email !== ''}]">
      <icon name="envelope" scale="1"></icon>
      <input
        type="email"
        name="email"
        placeholder="Email"
        aria-label="Email"
        id="email"
        :value="email"
        @input="(event) => updateSignupForm(event, 'email')"
        @blur="$v.email.$touch()">
    </div>
    <div :class="['password', {invalid: $v.password.$error && password !== ''}]">
      <icon name="unlock" scale="1"></icon>
      <input
        type="password"
        name="password"
        placeholder="Password"
        aria-label="Password"
        id="password"
        :value="password"
        @input="(event) => updateSignupForm(event, 'password')"
        @blur='$v.password.$touch()'>
    </div>
    <div :class="['confirmPassword', {invalid: $v.confirmPassword.$error && confirmPassword !== ''}]">
      <icon name="unlock" scale="1"></icon>
      <input
        type="password"
        name="password"
        placeholder="Retype Password"
        aria-label="Retype password"
        id="confirmPassword"
        :value="confirmPassword"
        @input="(event) => updateSignupForm(event, 'confirmPassword')"
        @blur='$v.confirmPassword.$touch()'>
    </div>
  </div>
</template>

<script>
import { required, email, minLength, sameAs } from 'vuelidate/lib/validators'

export default {
  computed: {
    username () {
      return this.$store.getters.signupForm.username
    },
    email () {
      return this.$store.getters.signupForm.email
    },
    password () {
      return this.$store.getters.signupForm.password
    },
    confirmPassword () {
      return this.$store.getters.signupForm.confirmPassword
    }
  },
  methods: {
    updateSignupForm (event, name) {
      this.$store.dispatch('updateSignupForm', {name, value: event.target.value})
    }
  },
  validations: {
    username: {
      required,
      minLen: minLength(5)
    },
    email: {
      required,
      email
    },
    password: {
      required,
      minLen: minLength(6)
    },
    confirmPassword: {
      sameAs: sameAs('password')
    }
  }
}
</script>

<style scoped>
.inputContainer {
  grid-area: loginInput;
  display: grid;
  grid-template-rows: 1fr 1fr 1fr 1fr;
  grid-template-areas:
    ' username '
    ' email '
    ' password '
    ' confirmPassword ';
  background-color: rgb(38, 49, 66);
}
.email {
  grid-area: email;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.username {
  grid-area: username;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.password {
  grid-area: password;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.confirmPassword {
  grid-area: confirmPassword;
  border-top: 1px solid rgba(250,250,250,0.5);
  border-bottom: 1px solid rgba(250,250,250,0.5);
}
input {
  height: 100%;
  width: 80%;
  padding-left: 10px;
  background-color: inherit;
  border: none;
  color: rgba(250,250,250,0.9);
}
.invalid input {
  color: #cb2431;
}
@media only screen and (min-width: 768px) {
  .inputContainer {
    border-radius: 10px;
  }
  .username {
    border-radius: 10px 10px 0px 0px;
  }
  .confirmPassword {
    border-radius: 0px 0px 10px 10px;
  }
}
</style>
