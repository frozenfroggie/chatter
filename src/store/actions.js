export default {
  updateMessage: ({commit}, payload) => {
    console.log('update!!!', payload)
    commit('updateMessage', payload)
  },
  addEmojiMessage: ({commit}, payload) => {
    console.log('update emoji!!!', payload)
    commit('addEmojiMessage', payload)
  },
  removeMessage: ({commit}) => {
    commit('removeMessage')
  },
  setSendMessageIntention: ({commit}, payload) => {
    commit('setSendMessageIntention', payload)
  },
  typingNotification: ({commit}, payload) => {
    commit('toogleIsTyping')
  },
  openEmojiPanel: ({commit}, payload) => {
    commit('setEmojiPanel', true)
  },
  closeEmojiPanel: ({commit}, payload) => {
    commit('setEmojiPanel', false)
  },
  resetState: ({commit}, payload) => {
    commit('resetState')
  }
}
