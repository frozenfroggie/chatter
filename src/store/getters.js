export default {
  messageText: state => state.messageText,
  addEmoji: state => state.addEmoji,
  emojiPanel: state => state.emojiPanel,
  sendMessageIntention: state => state.sendMessage.intention,
  isTyping: state => state.isTyping
}
