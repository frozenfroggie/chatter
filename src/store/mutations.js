export default {
  emailSend: (state) => {
    state.tokens.verificationToken.send = true
  },
  updateMessage: (state, payload) => {
    state.messageText = payload
  },
  addEmojiMessage: (state, payload) => {
    state.addEmoji = payload
    state.messageText += payload
  },
  removeMessage: (state) => {
    state.messageSend = true
  },
  sendMessagePending: (state, payload) => {
    state.sendMessage.pending = true
  },
  sendMessageSuccess: (state, payload) => {
    state.sendMessage.pending = false
    state.sendMessage.success = true
  },
  setSendMessageIntention: (state, payload) => {
    state.sendMessage.intention = payload
  },
  toogleIsTyping: (state) => {
    state.isTyping = !state.isTyping
  },
  setEmojiPanel: (state, payload) => {
    state.emojiPanel = payload
  },
  showVideoChatPrompt: (state) => {
    state.videoChat.isCalling = true
    state.videoChat.hangup = false
  },
  videoChatStart: (state) => {
    state.videoChat.isCalling = false
    state.videoChat.isStarted = true
    state.videoChat.hangup = false
  },
  videoChatHangup: (state) => {
    state.videoChat.isCalling = false
    state.videoChat.isStarted = false
    state.videoChat.hangup = true
  }
}
