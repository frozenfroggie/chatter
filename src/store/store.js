import Vue from 'vue'
import Vuex from 'vuex'
import VuexPersistence from 'vuex-persist'

import forms from './modules/forms'
import conversations from './modules/conversations'
import getMessages from './modules/getMessages'
import searchUsers from './modules/searchUsers'
import socket from './modules/socket'
import user from './modules/user'
import resizer from './modules/resizer'

import mutations from './mutations'
import getters from './getters'
import actions from './actions'

Vue.use(Vuex)

const vuexLocal = new VuexPersistence({
  storage: window.localStorage,
  supportCircular: true,
  modules: ['user', 'tokens', 'conversations']
})

export const store = new Vuex.Store({
  state: {
    socket: null,
    tokens: {
      accessToken: null,
      refreshToken: null,
      verificationToken: {
        send: false,
        confirmed: false
      }
    },
    messageText: '',
    addEmoji: '',
    emojiPanel: false,
    sendMessage: {
      intention: false,
      pending: false,
      success: false,
      error: false
    },
    isTyping: false,
    videoChat: {
      isCalling: false,
      isStarted: false,
      hangup: false
    }
  },
  getters,
  mutations,
  actions,
  modules: {
    forms,
    conversations,
    getMessages,
    searchUsers,
    socket,
    user,
    resizer
  },
  plugins: [vuexLocal.plugin]
})
