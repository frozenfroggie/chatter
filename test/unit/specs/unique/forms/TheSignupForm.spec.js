import Vuex from 'vuex'
import Vuelidate from 'vuelidate'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import TheSignupForm from '@/components/unique/forms/TheSignupForm';
import { store } from '@/store/store'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuelidate)

describe('TheSignupForm', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(TheSignupForm, {
      stubs: ['icon'],
      localVue,
      store
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
