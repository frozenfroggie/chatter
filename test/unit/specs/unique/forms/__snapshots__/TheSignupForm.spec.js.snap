// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSignupForm matches snapshot 1`] = `
<div class="inputContainer">
    <div class="username">
        <!---->
        <input type="text" name="username" placeholder="Username" aria-label="Username" id="username">
    </div>
    <div class="email">
        <!---->
        <input type="text" name="email" placeholder="Email" aria-label="Email" id="email">
    </div>
    <div class="password">
        <!---->
        <input type="password" name="password" placeholder="Password" aria-label="Password" id="password">
    </div>
    <div class="confirmPassword">
        <!---->
        <input type="password" name="password" placeholder="Retype Password" aria-label="Retype password" id="confirmPassword">
    </div>
</div>
`;
